### ISSUE TITLE
tile server my own test layer error

### ISSUE THREAD

**Issue by josephzhao** [tilezen/vector-datasource#1263]
* **What did you see?**
* **What did you expect to see?**
* **What map location are you having problems with?** _City and country are helpful, as well as tile coordinates or latitude / longitude._  
* **Screenshot?** _Props for animated gifs._

Hello All
  I meet a problem with the tile server, following is test..jinja2 for test layer:

SELECT
    '' AS name,
    t.* AS mz_properties,
    'openstreetmapdata.com' AS source,
    the_geom AS __geometry__,
    id AS __id__,
    NULL::boolean AS label_placement,
    NULL AS tunnel,
    NULL AS tags

FROM spatial_test t

my queries.yaml is:

all:
  - water
  - earth
  - test
  - places
  - landuse
  - roads
  - buildings
  - pois
  - boundaries
  - transit
layers:
  water:
    template: water.jinja2
    start_zoom: 0
    geometry_types: [Point, MultiPoint, Polygon, MultiPolygon, LineString, MultiLineString]
    simplify_before_intersect: true
    simplify_start: 9
    transform:
      - vectordatasource.transform.tags_create_dict
      - vectordatasource.transform.tags_name_i18n
      - vectordatasource.transform.tags_remove
      - vectordatasource.transform.water_tunnel
      - vectordatasource.transform.add_id_to_properties
      - vectordatasource.transform.detect_osm_relation
      - vectordatasource.transform.remove_feature_id
      - vectordatasource.transform.truncate_min_zoom_to_2dp
    sort: vectordatasource.sort.water
    area-inclusion-threshold: 1
  earth:
    template: earth.jinja2
    start_zoom: 0
    geometry_types: [Point, MultiPoint, Polygon, MultiPolygon, LineString, MultiLineString]
    simplify_before_intersect: true
    simplify_start: 9
    transform:
      - vectordatasource.transform.tags_create_dict
      - vectordatasource.transform.tags_name_i18n
      - vectordatasource.transform.tags_remove
      - vectordatasource.transform.detect_osm_relation
      - vectordatasource.transform.add_id_to_properties
      - vectordatasource.transform.remove_feature_id
      - vectordatasource.transform.truncate_min_zoom_to_2dp
    sort: vectordatasource.sort.earth
    area-inclusion-threshold: 1
  test:
    template: test.jinja2
    start_zoom: 10
    geometry_types: [Polygon, MultiPolygon]
    simplify_before_intersect: true
    simplify_start: 9
    area-inclusion-threshold: 1
  places:
    template: places.jinja2
    start_zoom: 0
    geometry_types: [Point, MultiPoint]
    transform:
      - vectordatasource.transform.tags_create_dict
      - vectordatasource.transform.tags_name_i18n
      - vectordatasource.transform.tags_remove
      - vectordatasource.transform.place_population_int
      - vectordatasource.transform.calculate_default_place_min_zoom
      - vectordatasource.transform.add_id_to_properties
      - vectordatasource.transform.detect_osm_relation
      - vectordatasource.transform.remove_feature_id
      - vectordatasource.transform.truncate_min_zoom_to_2dp
#    sort: vectordatasource.sort.places
    area-inclusion-threshold: 1
  landuse:
    template: landuse.jinja2
    start_zoom: 4
    geometry_types: [Polygon, MultiPolygon, LineString, MultiLineString]
    simplify_start: 4
    transform:
      - vectordatasource.transform.tags_create_dict
      - vectordatasource.transform.tags_name_i18n
      - vectordatasource.transform.tags_remove
      - vectordatasource.transform.normalize_tourism_kind
      - vectordatasource.transform.add_id_to_properties
      - vectordatasource.transform.detect_osm_relation
      - vectordatasource.transform.remove_feature_id
      - vectordatasource.transform.normalize_operator_values
      - vectordatasource.transform.truncate_min_zoom_to_2dp
    sort: vectordatasource.sort.landuse
    area-inclusion-threshold: 1
  roads:
    template: roads.jinja2
    start_zoom: 5
    geometry_types: [LineString, MultiLineString]
    simplify_start: 8
    transform:
      - vectordatasource.transform.tags_create_dict
      - vectordatasource.transform.tags_name_i18n
      - vectordatasource.transform.add_id_to_properties
      - vectordatasource.transform.detect_osm_relation
      - vectordatasource.transform.parse_layer_as_float
      - vectordatasource.transform.road_classifier
      - vectordatasource.transform.road_oneway
      - vectordatasource.transform.road_abbreviate_name
      - vectordatasource.transform.route_name
      - vectordatasource.transform.normalize_aerialways
      - vectordatasource.transform.normalize_cycleway
      - vectordatasource.transform.add_is_bicycle_related
      - vectordatasource.transform.extract_network_information
      - vectordatasource.transform.choose_most_important_network
      - vectordatasource.transform.road_trim_properties
      - vectordatasource.transform.remove_feature_id
      - vectordatasource.transform.tags_remove
      - vectordatasource.transform.truncate_min_zoom_to_2dp
    sort: vectordatasource.sort.roads
    area-inclusion-threshold: 1
  buildings:
    template: buildings.jinja2
    start_zoom: 13
    clip_factor: 3.0
    geometry_types: [Point, MultiPoint, Polygon, MultiPolygon]
    transform:
      - vectordatasource.transform.tags_create_dict
      - vectordatasource.transform.tags_name_i18n
      - vectordatasource.transform.tags_remove
      - vectordatasource.transform.add_id_to_properties
      - vectordatasource.transform.detect_osm_relation
      - vectordatasource.transform.parse_layer_as_float
      - vectordatasource.transform.building_height
      - vectordatasource.transform.building_min_height
      - vectordatasource.transform.synthesize_volume
      - vectordatasource.transform.normalize_tourism_kind
      - vectordatasource.transform.building_trim_properties
      - vectordatasource.transform.remove_feature_id
      - vectordatasource.transform.truncate_min_zoom_to_2dp
    sort: vectordatasource.sort.buildings
    area-inclusion-threshold: 1
  pois:
    template: pois.jinja2
    start_zoom: 4
    geometry_types: [Point, MultiPoint, Polygon, MultiPolygon]
    transform:
      - vectordatasource.transform.tags_create_dict
      - vectordatasource.transform.add_iata_code_to_airports
      - vectordatasource.transform.normalize_tourism_kind
      - vectordatasource.transform.normalize_social_kind
      - vectordatasource.transform.normalize_medical_kind
      - vectordatasource.transform.tags_name_i18n
      - vectordatasource.transform.add_uic_ref
      - vectordatasource.transform.tags_remove
      - vectordatasource.transform.add_id_to_properties
      - vectordatasource.transform.detect_osm_relation
      - vectordatasource.transform.remove_feature_id
      - vectordatasource.transform.remove_zero_area
      - vectordatasource.transform.make_representative_point
      - vectordatasource.transform.height_to_meters
      - vectordatasource.transform.pois_capacity_int
      - vectordatasource.transform.elevation_to_meters
      - vectordatasource.transform.normalize_operator_values
      - vectordatasource.transform.truncate_min_zoom_to_2dp
    sort: vectordatasource.sort.pois
    area-inclusion-threshold: 1
  boundaries:
    template: boundaries.jinja2
    start_zoom: 0
    geometry_types: [Polygon, MultiPolygon, LineString, MultiLineString]
    simplify_before_intersect: true
    simplify_start: 8
    transform:
      - vectordatasource.transform.tags_create_dict
      - vectordatasource.transform.tags_name_i18n
      - vectordatasource.transform.tags_remove
      - vectordatasource.transform.admin_level_as_int
      - vectordatasource.transform.add_id_to_properties
      - vectordatasource.transform.detect_osm_relation
      - vectordatasource.transform.remove_feature_id
      - vectordatasource.transform.truncate_min_zoom_to_2dp
    area-inclusion-threshold: 1
  transit:
    template: transit.jinja2
    start_zoom: 5
    geometry_types: [LineString, MultiLineString, Polygon, MultiPolygon]
    transform:
      - vectordatasource.transform.tags_create_dict
      - vectordatasource.transform.tags_name_i18n
      - vectordatasource.transform.add_uic_ref
      - vectordatasource.transform.tags_remove
      - vectordatasource.transform.add_id_to_properties
      - vectordatasource.transform.detect_osm_relation
      - vectordatasource.transform.route_name
      - vectordatasource.transform.parse_layer_as_float
      - vectordatasource.transform.remove_feature_id
      - vectordatasource.transform.truncate_min_zoom_to_2dp
    sort: vectordatasource.sort.transit
    area-inclusion-threshold: 1
post_process:
  - fn: vectordatasource.transform.numeric_min_filter
    params:
      source_layer: buildings
      mode: any
      filters:
        13: { area: 1600, volume: 300000 }
        14: { area:  100, volume: 200000 }
        15: { area:  100, volume: 100000 }

  - fn: vectordatasource.transform.build_fence
    params:
      base_layer: landuse
      start_zoom: 16
      prop_transform:
        kind: true
        id: true
        source: true
        area: true
        min_zoom: true
  - fn: vectordatasource.transform.drop_properties
    params:
      source_layer: landuse
      start_zoom: 4
      properties: [barrier]

  # sort key
  - fn: vectordatasource.transform.csv_match_properties
    resources:
      matcher:
        type: file
        init_fn: vectordatasource.transform.CSVMatcher
        path: spreadsheets/sort_rank/landuse.csv
    params:
      source_layer: landuse
      target_value_type: int
  - fn: vectordatasource.transform.csv_match_properties
    resources:
      matcher:
        type: file
        init_fn: vectordatasource.transform.CSVMatcher
        path: spreadsheets/sort_rank/roads.csv
    params:
      source_layer: roads
      target_value_type: int
  - fn: vectordatasource.transform.csv_match_properties
    resources:
      matcher:
        type: file
        init_fn: vectordatasource.transform.CSVMatcher
        path: spreadsheets/sort_rank/earth.csv
    params:
      source_layer: earth
      target_value_type: int
  - fn: vectordatasource.transform.csv_match_properties
    resources:
      matcher:
        type: file
        init_fn: vectordatasource.transform.CSVMatcher
        path: spreadsheets/sort_rank/boundaries.csv
    params:
      source_layer: boundaries
      target_value_type: int
  - fn: vectordatasource.transform.csv_match_properties
    resources:
      matcher:
        type: file
        init_fn: vectordatasource.transform.CSVMatcher
        path: spreadsheets/sort_rank/buildings.csv
    params:
      source_layer: buildings
      target_value_type: int
  - fn: vectordatasource.transform.csv_match_properties
    resources:
      matcher:
        type: file
        init_fn: vectordatasource.transform.CSVMatcher
        path: spreadsheets/sort_rank/transit.csv
    params:
      source_layer: transit
      target_value_type: int
  # this needs to run before the water sort_rank csv matcher
  - fn: vectordatasource.transform.exterior_boundaries
    params:
      base_layer: water
      start_zoom: 9
      prop_transform:
        kind: true
        id: true
        source: true
        boundary: [true]
        area: true
      snap_tolerance: 0.125
  # have to do the water properties matching _after_ exterior boundaries
  # as it depends on having the "boundary: yes" property available.
  - fn: vectordatasource.transform.csv_match_properties
    resources:
      matcher:
        type: file
        init_fn: vectordatasource.transform.CSVMatcher
        path: spreadsheets/sort_rank/water.csv
    params:
      source_layer: water
      target_value_type: int

  # assign `scale_rank`
  - fn: vectordatasource.transform.csv_match_properties
    resources:
      matcher:
        type: file
        init_fn: vectordatasource.transform.CSVMatcher
        path: spreadsheets/scale_rank/buildings.csv
    params:
      source_layer: buildings
      target_value_type: int

  # drop all polygonal features that don't meet the minimum area
  # threshold. this depends on the mz_min_pixels property to be set on
  # features first
  - fn: vectordatasource.transform.drop_features_mz_min_pixels
    params:
      property: mz_min_pixels
      source_layers:
        # all layers that have polygonal geometries
        - boundaries
        - buildings
        - earth
        - landuse
        - transit
        - water

  - fn: vectordatasource.transform.drop_properties
    params:
      source_layer: roads
      properties: [layer]
  - fn: vectordatasource.transform.overlap
    params:
      base_layer: buildings
      cutting_layer: landuse
      attribute: kind
      target_attribute: landuse_kind
      cutting_attrs: { sort_key: 'sort_rank', reverse: True }
  - fn: vectordatasource.transform.admin_boundaries
    params:
      base_layer: boundaries
      start_zoom: 8

  # drop names on boundary lines (country, region, macroregion) except zoom 7 from Natural Earth
  - fn: vectordatasource.transform.drop_properties
    params:
      source_layer: boundaries
      start_zoom: 0
      end_zoom: 6
      properties: [name]

  - fn: vectordatasource.transform.handle_label_placement
    params:
      layers:
        - water
        - earth
      location_property: mz_label_placement
      label_property_name: label_placement
      label_property_value: true
      label_where: >-
        'name' in properties

  - fn: vectordatasource.transform.handle_label_placement
    params:
      layers:
        - landuse
      start_zoom: 15
      location_property: mz_label_placement
      label_property_name: label_placement
      label_property_value: true
      label_where: >-
        'name' in properties or
        ('sport' in properties and properties.get('kind', 'rock') not in ('rock', 'stone'))

  # drop mz_label placement at zooms we don't apply handle_label_placement
  - fn: vectordatasource.transform.drop_properties
    params:
      source_layer: landuse
      start_zoom: 0
      end_zoom: 14
      properties: [mz_label_placement]
      geom_types: [Polygon, MultiPolygon]

  - fn: vectordatasource.transform.handle_label_placement
    params:
      layers:
        - buildings
      start_zoom: 16
      location_property: mz_label_placement
      label_property_name: label_placement
      label_property_value: true
      label_where: >-
        'name' in properties

  # drop mz_label placement at zooms we don't apply handle_label_placement
  - fn: vectordatasource.transform.drop_properties
    params:
      source_layer: buildings
      start_zoom: 0
      end_zoom: 15
      properties: [mz_label_placement]
      geom_types: [Polygon, MultiPolygon]

  - fn: vectordatasource.transform.remove_duplicate_features
    params:
      source_layer: landuse
      end_zoom: 15
      property_keys: [name, kind]
      geometry_types: [Point]
      min_distance: 256.0

  # drop the names and less important additional tags of small landuse polygons
  - fn: vectordatasource.transform.drop_properties
    params:
      source_layer: landuse
      start_zoom: 4
      end_zoom: 12
      properties: [name, sport, religion, surface]
      where: >-
        pixel_area > area
      geom_types: [Polygon, MultiPolygon]

  - fn: vectordatasource.transform.drop_features_where
    params:
      source_layer: water
      start_zoom: 3
      where: >-
        'label_placement' not in properties and kind in ('sea', 'bay', 'strait', 'fjord')
  - fn: vectordatasource.transform.drop_features_where
    params:
      source_layer: earth
      start_zoom: 0
      where: >-
        'label_placement' not in properties and kind in
        ('archipelago', 'island', 'islet', 'continent', 'valley', 'ridge')

  # merge water and earth polygons, see https://github.com/tilezen/vector-datasource/issues/1106
  - fn: vectordatasource.transform.merge_polygon_features
    params:
      source_layer: earth
  - fn: vectordatasource.transform.merge_polygon_features
    params:
      source_layer: water

  - fn: vectordatasource.transform.generate_address_points
    params:
      source_layer: buildings
      start_zoom: 16
  - fn: vectordatasource.transform.remove_duplicate_features
    params:
      source_layer: buildings
      end_zoom: 15
      property_keys: [name, kind]
      geometry_types: [Point]
      min_distance: 256.0
  - fn: vectordatasource.transform.remove_duplicate_features
    params:
      source_layer: pois
      end_zoom: 15
      property_keys: [name, kind]
      geometry_types: [Point]
      min_distance: 256.0
  - fn: vectordatasource.transform.drop_features_where
    params:
      source_layer: landuse
      start_zoom: 6
      where: >-
        properties.get('mz_is_building') is True and
        properties.get('label_placement') is None
  - fn: vectordatasource.transform.merge_duplicate_stations
    params:
      source_layer: pois
      end_zoom: 15
  - fn: vectordatasource.transform.normalize_station_properties
    params:
      source_layer: pois
  - fn: vectordatasource.transform.rank_features
    params:
      source_layer: pois
      items_matching:
        kind: station
      rank_key: kind_tile_rank
  - fn: vectordatasource.transform.rank_features
    params:
      source_layer: places
      items_matching:
        kind: [neighbourhood, microhood, macrohood]
      rank_key: kind_tile_rank
      start_zoom: 11
  - fn: vectordatasource.transform.rank_features
    params:
      source_layer: pois
      items_matching:
        kind: [peak, volcano]
      rank_key: kind_tile_rank
      start_zoom: 9
  - fn: vectordatasource.transform.remove_duplicate_features
    params:
      source_layers: [pois, landuse, buildings]
      end_zoom: 15
      property_keys: [name]
      geometry_types: [Point]
      min_distance: 64.0
  - fn: vectordatasource.transform.drop_properties
    params:
      source_layer: roads
      start_zoom: 0
      end_zoom: 14
      properties: [is_bridge, is_tunnel]
      where: >-
        (kind == 'path' and zoom < 15) or
        (kind in ['minor_road', 'major_road', 'highway', 'rail'] and zoom < 13)
  # drop certain road properties at lower zooms
  - fn: vectordatasource.transform.drop_properties
    params:
      source_layer: roads
      start_zoom: 0
      end_zoom: 14
      properties: [oneway, crossing, sidewalk, sidewalk_left, sidewalk_right]
  - fn: vectordatasource.transform.drop_properties
    params:
      source_layer: roads
      start_zoom: 0
      end_zoom: 14
      properties: [name, ref, network, shield_text, all_networks, all_shield_texts]
      where: >-
        (kind == 'rail' and zoom < 15) or
        (kind == 'minor_road' and zoom < 14) or
        (kind == 'major_road' and zoom <  7)
  # this is a patch because we still want to drop name, network from major_road
  # features between zoom 7 and 10
  - fn: vectordatasource.transform.drop_properties
    params:
      source_layer: roads
      start_zoom: 7
      end_zoom: 10
      properties: [name, all_networks, all_shield_texts]
      where: >-
        kind == 'major_road'
  # this is a patch to get rid of name, but keep ref & network, for highways
  # when zoom < 11.
  - fn: vectordatasource.transform.drop_properties
    params:
      source_layer: roads
      start_zoom: 7
      end_zoom: 10
      properties: [name]
      where: >-
        kind == 'highway'
  # drop name, ref and the multi-shield properties, but keep single-shield
  # properties, when z<=6 (i.e: NE data)
  - fn: vectordatasource.transform.drop_properties
    params:
      source_layer: roads
      start_zoom: 0
      end_zoom: 6
      properties: [name, ref, all_networks, all_shield_texts]
      where: >-
        kind == 'highway'
  - fn: vectordatasource.transform.update_parenthetical_properties
    params:
      source_layer: pois
      values: ['closed', 'historical']
      target_min_zoom: 17
      drop_below_zoom: 16
  - fn: vectordatasource.transform.update_parenthetical_properties
    params:
      source_layer: buildings
      values: ['closed', 'historical']
      target_min_zoom: 17
      drop_below_zoom: 16
  - fn: vectordatasource.transform.merge_line_features
    params:
      source_layer: transit
      start_zoom: 0
      end_zoom: 15
  # only keep all stations below z12, then cut it down to the top few.
  - fn: vectordatasource.transform.keep_n_features
    params:
      source_layer: transit
      start_zoom: 9
      end_zoom: 11
      items_matching: { kind: station }
      max_items: 20
  - fn: vectordatasource.transform.keep_n_features
    params:
      source_layer: transit
      start_zoom: 12
      end_zoom: 13
      items_matching: { kind: station }
      max_items: 30
  - fn: vectordatasource.transform.keep_n_features
    params:
      source_layer: pois
      start_zoom: 9
      end_zoom: 15
      items_matching: { kind: [peak, volcano] }
      max_items: 5
  - fn: vectordatasource.transform.drop_properties_with_prefix
    params: {prefix: mz_}

  - fn: vectordatasource.transform.merge_building_features
    params:
      source_layer: buildings
      start_zoom: 13
      end_zoom: 14
      quantize:
        13: vectordatasource.transform.quantize_height_round_nearest_10_meters
        14: vectordatasource.transform.quantize_height_round_nearest_5_meters
      drop:
        - name
        - addr_housenumber
        - addr_street
      max_merged_features: 1000

  - fn: vectordatasource.transform.drop_small_inners
    params:
      end_zoom: 13
      source_layers: [landuse, water]
      pixel_area: 0.1

  - fn: vectordatasource.transform.simplify_and_clip
    params: {simplify_before: 16}

  - fn: vectordatasource.transform.intercut
    params:
      base_layer: roads
      cutting_layer: landuse
      attribute: kind
      target_attribute: landuse_kind
      cutting_attrs: { sort_key: 'sort_rank', reverse: True }
  # merge any remaining polygons by their properties.
  - fn: vectordatasource.transform.merge_polygon_features
    params:
      source_layer: landuse
      start_zoom: 4
      end_zoom: 12

  - fn: vectordatasource.transform.merge_line_features
    params:
      source_layer: roads
      start_zoom: 8
      end_zoom: 15

  - fn: vectordatasource.transform.buildings_unify
    params:
      source_layer: buildings
      start_zoom: 15

my config.yaml is :

postgresql:
  host: localhost
  port: 5432
  # multiple databases can be specified, and these are iterated
  # through to balance query loads. This is useful when connecting to
  # pgbouncer, which can dispatch to different back end databases
  # based on the name.
  dbnames: [osm]
  user: jzhao
  password: jzhao
queries:
  config: ../vector-datasource/queries.yaml
  template-path: ../vector-datasource/queries
  # whether to reload the jinja query templates. This should be off in
  # production.
  reload-templates: true

# format extensions to support
# buffered Mapbox Vector Tiles are also possible by specifying mvtb
formats: [json, topojson, mvt]
# buffer configuration, see tilequeue sample config for details
buffer: {}

server:
  host: 0.0.0.0
  port: 8080
  debug: true
  reload: true
  threaded: false

http:
  # whether to include cors headers on responses
  cors: true
  # client cache time in seconds ("Cache-Control: max-age" header)
  max-age: 300

# optional for health checks. This can be useful for monitoring to
# verify that the service is still running and can connect to the
# database.
health:
  # request path to listen for health checks
  url: /_health

cache:
   type: file
#   redis:
#     url: redis://localhost:6379
#     options:
#       # prefix added onto saved keys for namespace purposes
#       key_prefix: tiles
#       # time in seconds to keep key in cache
#       expires: 900
   file:
     prefix: tiles_data

# to cache tiles locally, enable a cache. This can be useful when
# experiencing timeouts for low or mid zoom ranges.
# cache:
#   type: <redis|file|null, default 'null'>
#   redis:
#     url: redis://localhost:6379
#     options:
#       # prefix added onto saved keys for namespace purposes
#       key_prefix: tiles
#       # time in seconds to keep key in cache
#       expires: 900
#   file:
#     prefix: directory_prefix

# to support requesting tiles for different metatile sizes
# these should get prefixed to the beginning of the url path
path_tile_size:
  # NOTE: the keys should be strings
  "256": 1
  "512": 2

# requests for zoom levels higher than this will 404
max_interesting_zoom: 20

for example data and layer is ok

but for my data show errors as following:

shapely.geos.ReadingError

ReadingError: Could not create geometry because of errors while reading input.
Traceback (most recent call last)

    File "/opt/tileserver/tileserver/__init__.py", line 148, in __call__

    response = self.handle_request(request)

    File "/opt/tileserver/tileserver/__init__.py", line 232, in handle_request

    self.post_process_data,

    File "/usr/local/lib/python2.7/dist-packages/tilequeue/process.py", line 220, in process_coord_no_format

    shape = loads(wkb)

    File "/usr/local/lib/python2.7/dist-packages/shapely/wkb.py", line 16, in loads

    return reader.read(data)

    File "/usr/local/lib/python2.7/dist-packages/shapely/geos.py", line 363, in read

    raise ReadingError("Could not create geometry because of errors "

    ReadingError: Could not create geometry because of errors while reading input.

The debugger caught an exception in your WSGI application. You can now look at the traceback which led to the error.

To switch between the interactive traceback and the plaintext one, you can click on the "Traceback" headline. From the text traceback you can also create a paste of it. For code execution mouse-over the frame you want to debug and click on the console icon on the right side.

You can execute arbitrary Python code in the stack frames and there are some extra helpers available for introspection:

    dump() shows all variables in the frame
    dump(obj) dumps all that's known about the object

**Comment by nvkelso** [comment]
Hi @ josephzhao. It's a little hard to follow your issue description. Do you have a repo we can check out with the original (your modifications) files? 

Confirm your basic Tilezen setup is working:

- If you remove the `test.jinja` from your config (just using the base Tilezen layers) are tiles returned like expected? 

Once you confirm the basic setup works, let's move on to your `test.jinja`...

- Are you sure your geometries are in Web Mercator projection and valid shapes?

**Comment by josephzhao** [comment]
Hi Nathaniel
Thank you so much for your replyThe basic setup as those test layers are ok, I can download different format tile data,  so I uploaded my geometry test table that is EPSG: 4326, and I created other geometry column EPSG:3857 and EPSG:900913 and copied the sample layer seeting, but I don't know where is the problem?
Thanks
Joseph

**Comment by rmarianski** [comment]
The error message looks like the geometry it's trying to read is just not wkb. I'd suggest inspecting what the value actually is.

**Comment by nvkelso** [comment]
Closing for in-activity.
